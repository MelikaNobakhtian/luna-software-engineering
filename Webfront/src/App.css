.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}
* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: "PT Sans", sans-serif;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.App-color1 {
  background-color: #f0f3bd;
}
.App-color2 {
  background-color: #02c39a;
}
.App-color3 {
  background-color: #00a896;
}
.App-color4 {
  background-color: #028090;
}
.App-color5 {
  background-color: #05668d;
}
