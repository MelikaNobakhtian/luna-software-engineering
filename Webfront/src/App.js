import React from "react";
import "./App.css";
//import "bootstrap/dist/css/bootstrap.min.css"
import Navbar from "./Navbar.js";
//import {Container ,Row,Col,Button,nav} from  "react-bootstrap";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  //Link,
  Redirect,
} from "react-router-dom";
//import '../node_modules/bootstrap/dist/css/bootstrap.min.css';
import login from "./components/login/login";
import signUp from "./components/login/signUp";
import editProfileDoctor from "./components/editProfile-doctor/editProfile-doctor";
import UserProfile from "./components/userProfile/userProfile";
import ForgotPassword from "./components/login/forgotPassword";
import Verification from "./components/login/verification";
import Doctorcalender from "./components/doctortimes/Doctorcalender";
//import Cookies from "js-cookie";
import Home from "./components/home/home";
import SearchResult from "./components/search/searchResult"

function App() {
  function PrivateRoute({ component: Component, ...rest }) {
    return (
      <Route
        {...rest}
        render={(props) =>
          // !!Cookies.get("userTokenA")
          true ? (
            <div>
              <Navbar></Navbar>
              <Component {...props} />
            </div>
          ) : (
            <Redirect
              to={{ pathname: "/login", state: { from: props.location } }}
            />
          )
        } //or from: props.location
      />
    );
  }
  return (
    <Router>
      <div data-testid="router" className="text-right" style={{ direction: "rtl" }}>
        <Switch>
          <Route path="/" exact={true}>
            <Navbar />
            <Home />
          </Route>

          <Route path="/signup" component={signUp} />
          <Route path="/login" component={login} />

          <PrivateRoute path="/doctorProfile" component={editProfileDoctor} />
          <PrivateRoute path="/userProfile" component={UserProfile} />

          <Route path="/forgotPassword/:tokenId/:token"component={ForgotPassword}/>
          <Route path="/verification/:tokenId" component={Verification} />

          <Route path="/doctorcal">
            <Navbar />
            <Doctorcalender />
          </Route>

          <Route path="/searchResult/:search">
            <Navbar />
            <SearchResult />
          </Route>



          <Route path="*">
              <div class="alert alert-warning" role="alert">
                صفحه وجود ندارد
              </div>
            </Route>
        </Switch>
      </div>
    </Router>
  );
}

export default App;
