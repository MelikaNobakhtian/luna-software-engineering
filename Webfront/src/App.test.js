import { render, screen, fireEvent } from "@testing-library/react";
import App from "./App";
import React from "react";
import { Link, Router } from "react-router-dom";
import { createMemoryHistory } from "history";

test("renders learn react link", () => {
  render(<App />);
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});

// it('should take a snapshot', () => {
//   const { asFragment } = render(<App />)

//   expect(asFragment(<App />)).toMatchSnapshot()
//  })

const renderWithRouter = (component) => {
  const history = createMemoryHistory();
  return {
    ...render(<Router history={history}>{component}</Router>),
  };
};

it("should render the navbar", () => {
  const { container, getByTestId } = renderWithRouter(<App />);
  const navbar = getByTestId("navbar");
  const router = getByTestId("router");
  //const searchResult = getByTestId("searchResult");
  const home = getByTestId("home");

  //expect(container.innerHTML).toMatch("Home page");
  expect(router).toContainElement(navbar);
});

it("should render the home page", () => {
  const { container, getByTestId } = renderWithRouter(<App />);
  const router = getByTestId("router");
  const home = getByTestId("home");

  expect(router).toContainElement(home);
});

it("should navigate to the searchResult", () => {
  const { container, getByTestId } = renderWithRouter(<App />);
  fireEvent.click(getByTestId("goToSearch"));
  const router = getByTestId("router");
  const searchResult = getByTestId("searchResult");
  const navbar = getByTestId("navbar");

  expect(router).toContainElement(searchResult);
  expect(router).toContainElement(navbar);
});


