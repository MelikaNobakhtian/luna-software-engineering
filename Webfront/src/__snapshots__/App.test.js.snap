// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-right"
    style="direction: rtl;"
  >
    <div
      class=""
      style="background-color: rgb(235, 252, 255);"
    >
      <nav
        class="navbar navbar-expand-lg navbar-light"
        dir="rtl"
      >
        <div
          class="container-fluid"
        >
          <div
            class="navbar-brand"
            style="color: rgb(2, 128, 144); font-weight: bolder;"
          >
            پزشک
            <button
              aria-controls="offcanvasRight"
              class="btn mx-4 round-3 shadow-0"
              data-bs-target="#offcanvasRight"
              data-bs-toggle="offcanvas"
              style="background-color: white; border-radius: 100px; border-color: lightgray;"
            >
              <svg
                aria-controls="offcanvasRight"
                class="round-3 mb-1 align-self-center justify-self-center"
                color="gray"
                data-bs-target="#offcanvasRight"
                data-bs-toggle="offcanvas"
                dir="rtl"
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                style="color: gray;"
                viewBox="0 0 16 16"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
          <button
            aria-controls="navbarNav"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class=" navbar-toggler"
            data-bs-target="#navbarNav"
            data-bs-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
              data-bs-target=".navbar-collapse.show"
            />
          </button>
          <div
            class="collapse navbar-collapse mb-5 mb-lg-0"
            id="navbarNav"
          >
            <ul
              class="navbar-nav"
            >
              <li
                class="btn nav-item nav-link ms-auto"
              >
                گرفتن نوبت
              </li>
            </ul>
            <form
              class="d-flex col-lg-8  col-md-6 mt-lg-auto mb-lg-auto mb-n2 mt-3 col-11 round-3 me-lg-5 me-2"
            />
          </div>
          <div
            class="collapse navbar-collapse flex-row-reverse"
            id="navbarNav"
          >
            <ul
              class="navbar-nav  "
            >
              <div
                class="nav-item mt-lg-auto mb-lg-auto mb-2 mt-n5  ms-5"
              >
                <div
                  class="btn p-1 "
                >
                  <svg
                    class="ms-2"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 100-6 3 3 0 000 6z"
                      fill-rule="evenodd"
                    />
                  </svg>
                  ورود یا ثبت نام
                </div>
              </div>
              <li
                class="btn nav-item nav-link pull-left ms-auto"
                style="color: rgb(0, 168, 150); font-weight: bold;"
              >
                <small>
                  مشاهده ی اتاق ها
                </small>
              </li>
            </ul>
          </div>
        </div>
      </nav>
      <div
        aria-labelledby="offcanvasRightLabel"
        class="offcanvas offcanvas-end"
        id="offcanvasRight"
        style="background-color: rgb(227, 238, 238);"
        tabindex="-1"
      >
        <div
          class="offcanvas-header"
        >
          <button
            aria-label="Close"
            class="btn-close text-reset"
            data-bs-dismiss="offcanvas"
            type="button"
          />
          <h5
            id="offcanvasRightLabel"
          >
            جستجو
          </h5>
        </div>
        <div
          class="offcanvas-body m-1 "
        >
          <p
            class=""
          >
            قسمت های مورد نظر خود را پر کنید
          </p>
          <form
            class="row g-3"
          >
            <div
              class="col-6"
            >
              <input
                class="form-control "
                id="name"
                placeholder="نام دکتر..."
                type="text"
                value=""
              />
            </div>
            <div
              class="col-6"
            >
              <input
                class="form-control "
                id="lastname"
                placeholder="فامیلی دکتر..."
                type="text"
                value=""
              />
            </div>
            <div
              class="col-12"
            >
              <input
                aria-label=""
                class="form-control  "
                id="specialty"
                placeholder="تخصص..."
                type="text"
                value=""
              />
            </div>
            <div
              class="col-6"
            >
              <input
                aria-label=""
                class="form-control"
                id="state"
                placeholder="استان..."
                type="text"
                value=""
              />
            </div>
            <div
              class="col-6"
            >
              <input
                aria-label=""
                class="form-control"
                id="city"
                placeholder="شهر..."
                type="text"
                value=""
              />
            </div>
            <div
              class="btn btn-primary col-4 mx-2"
            >
              بگرد
            </div>
          </form>
        </div>
      </div>
    </div>
    <div>
      home
    </div>
  </div>
</DocumentFragment>
`;
